### ISSUE TITLE
Review: Frostpunk 2

### ISSUE THREAD

**Issue by ChemGuy1611** [Nexus-Mods/Vortex#16447]
### Nexus Username

ChemBoy1

### Extension URL

https://www.nexusmods.com/site/mods/960

### Game URL

https://www.nexusmods.com/frostpunk2

### Existing Extension URL

NONE

### New features

This is based on a standard UE5 mod installation model. Supports all game store versions of the game, including Game Pass.

There is a modding SDK, FrostKit, that will be released soon. There may be additional or alternate ways to install mods when that comes out. I will update the extension to accommodate this. 

### Information

- [X] I confirm the above is accurate

### Packaging

- [X] This extension is packaged correctly

### Testing

- [X] This game extension has been tested
### Review Tasks

If a task fails, contact the author to request changes before continuing.

- [ ] Double-check for existing extension
- [ ] Is the extension [packaged correctly](https://github.com/Nexus-Mods/Vortex/wiki/How-to-package-a-game-extension)?
- [ ] Does it install into Vortex?
- [ ] Does it correctly discover the game?
- [ ] Does it successfully install a mod?
- [ ] Does it successfully install a collection?
- [ ] Does the game run correctly with the mods installed? ​

When reviewed and passed, please complete the following tasks:

- [ ] Run the GitHub Actions to add to manifest
- [ ] Contact author
- [ ] Ask Community to enable the Vortex button for the game
- [ ] Update the #vortex-announcements channel on Discord

**Comment by IDCs** [comment]
Hey there @ChemGuy1611,

As usual, thank you for your contributions.

I can see that you're building up the `requiredFiles` array while querying the executable - please note that's not going to work, by the time the executable functor gets queried, the game registration call had already been processed. And the required files array had already been set in the state.

Fortunately the array had been set to be empty so it will not block users from managing the game as long as auto-discovery works (appears to be fine at a glance, but I can't test it on my end).

This is more of a heads up for the future as I will approve the extension.

**Comment by ChemGuy1611** [comment]
Yes, I do understand what you are saying. I learned about this method from insomnius's Palworld extension. For the executable and required files, this uses a basically identical method to that extension. I don't know enough about the API to understand exactly why this structure works, but I have tested it with quite a few games now that I went back and updated to integrate Xbox version support into, and it does work well. It seems that having the required files array empty until it is filled with the executable is necessary. It took me a loooong time to discover this fact.

So basically if you choose another common file to put in required files, the getExecutable function won't work properly. The array needs to remain empty until filled by the discovered executable.

**Comment by IDCs** [comment]
The problem with not providing a file in the requiredFiles array is that you can manually set the game folder to any directory (even `C:/Windows/System32/`) and Vortex will allow it as it can't check if the directory the user is setting is actually the game or not. It also may fail when the user chooses to unmanage the game (as again, Vortex can't check if the game is even installed)

It's not a big deal and as you said there are several game extensions that do not provide a populated array due Vortex's limitation of not being able to provide this array dynamically. But it's usually practiced when the game directory structure differs greatly between different game stores.

Palworld for example has a Win64 directory for Steam and all other non-xbox game stores; while xbox expects the files to be located inside WinGDK. That's not something that Vortex can support currently which is why that extension has an empty array.

The Frostpunk 2 directory structure for xbox is almost identical to the other game stores though and they do include common files that can be tested against.

I did notice that some of the game extensions you wrote had the same pattern, but assumed that the directory structure is different between game stores.

I'll add your extension to the manifest, but please be aware that you should have that array populated on Vortex start-up unless the game structure is widely different like in the case of Palworld.

**Comment by IDCs** [comment]
Added - thanks again.

**Comment by ChemGuy1611** [comment]
I did a quick test and it seems like there is no problem with adding another common file to requiredFiles that is not the executable. I will update those extensions I listed to include this just to stop users from setting random folders as the game folder.
